sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LogicalPlan.scala

```diff
@@ -60,7 +60,8 @@ abstract class LogicalPlan
    * [[org.apache.spark.sql.catalyst.analysis.UnresolvedRelation UnresolvedRelation]]
    * should return `false`).
    */
-  lazy val resolved: Boolean = expressions.forall(_.resolved) && childrenResolved
+  lazy val resolved: Boolean = expressions.forall(_.resolved) && childrenResolved &&
+    missingInput.isEmpty
```

Comment:
`missingInput` is special, mostly we can't resolve it. I think that's why we didn't consider it in the `resolved` at the first place.

We can update the if condition in `ResolveMissingReferences` to take `missingInput` into consideration.

Reply:
Yeah, I found that this change causes one test failure.